homeplug_exporter.go

```diff
@@ -327,12 +328,29 @@ ChanLoop:
 	for {
 		select {
 		case n := <-ch:
+			addr := n.Address.String()
+			if seen[addr] {
+				continue
+			}
 			ni = append(ni, n)
+			seen[addr] = true
+			// Query each remote station directly.
+			for _, station := range n.Stations {
+				err := write_homeplug(iface, conn, station.Address)
+				if err != nil {
+					return nil, fmt.Errorf("write_homeplug failed: %v", err)
```

Comment:
I tend to use `%w` rather than `%v`, so that the caller can unwrap your error to get at the original (wrapped) error.
```suggestion
					return nil, fmt.Errorf("write_homeplug failed: %w", err)
```

Reply:
I did not know this one either! I should apply this to the whole file!